setuptools/tests/config/test_setupcfg.py

```diff
@@ -85,24 +85,24 @@ def test_basic(self, tmpdir):
             '[options]\n'
             'scripts = bin/a.py, bin/b.py\n',
         )
-        config_dict = read_configuration('%s' % config)
+        config_dict = read_configuration(f'{config}')
         assert config_dict['metadata']['version'] == '10.1.1'
         assert config_dict['metadata']['keywords'] == ['one', 'two']
         assert config_dict['options']['scripts'] == ['bin/a.py', 'bin/b.py']
 
     def test_no_config(self, tmpdir):
         with pytest.raises(DistutilsFileError):
-            read_configuration('%s' % tmpdir.join('setup.cfg'))
+            read_configuration('{}'.format(tmpdir.join('setup.cfg')))
```

Comment:
The format call here seems quite extra
```suggestion
            read_configuration(str(tmpdir.join('setup.cfg')))
```
(Can't double check from my phone, but if `tmpdir.join` already returns a `str`):

```suggestion
            read_configuration(tmpdir.join('setup.cfg'))
```

Reply:
According to the [pytest documentation](https://docs.pytest.org/en/8.2.x/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures):
> The `tmpdir` and `tmpdir_factory` fixtures are similar to `tmp_path` and `tmp_path_factory`, but use/return legacy [`py.path.local`](https://py.readthedocs.io/en/latest/path.html) objects rather than standard [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html#pathlib.Path) objects.

```python
>>> import py.path
>>> 
>>> tmpdir = py.path.local("/tmp")
>>> x = tmpdir.join('setup.cfg')
>>> 
>>> type(x)
<class 'py._path.local.LocalPath'>
>>> 
>>> str(x)
'/tmp/setup.cfg'
>>> 
>>> x
local('/tmp/setup.cfg')
>>> 
```